### PR TITLE
prov/mlx: Fix build issue

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
Signed-off-by: Dmitry Gladkov <dmitry.gladkov@intel.com>

### CODE REVIEW

File: prov/mlx/src/mlx_init.c [hunk 1]
```diff
@@ -119,7 +118,13 @@ struct fi_fabric_attr mlx_fabric_attrs = {
 struct fi_ep_attr mlx_ep_attrs = {
 	.type = FI_EP_RDM,
 	.protocol = FI_PROTO_MLX,
-	.protocol_version = UCP_API_RELEASE,
+#if defined(UCP_API_RELEASE) && (UCP_API_RELEASE <= 2947)
+#warning "HPCX 1.9.7 have an issue with UCP_API_VERSION macro"
+	.protocol_version = (((UCP_API_MAJOR) << UCP_VERSION_MAJOR_SHIFT)|
+        ((UCP_API_MINOR) << UCP_VERSION_MINOR_SHIFT)),
+#else
+	.protocol_version = (UCP_API_VERSION),
```
Comment (shefty) [inline prov/mlx/src/mlx_init.c]: Isn't this using UCP_API_VERSION when it is NOT defined?

**Comment by shefty** [comment]
See PR #3252 and #3248.  @gladkovdmitry17 - Can you please merge any changes from the other pull requests that are needed into this one?

File: prov/mlx/src/mlx_init.c (hunk 1 above)
Comment (gladkovdmitry17) [reply prov/mlx/src/mlx_init.c]: Seems, [UCP_API_VERSION](https://github.com/openucx/ucx/blob/32869e6f8816bc0d8d34dc2eff77a1a287e7860d/src/ucp/api/ucp_version.h.in#L22) macro is always defined. But `UCP_API_RELEASE` is defined in `ucp/api/ucp_version.h` for release version of UCX.

**Comment by gladkovdmitry17** [comment]
@shefty Sure, it's done. But if I apply #3248 it currently fails due to there are no fixes for the compilation failure with clang compiler.

**Comment by gladkovdmitry17** [comment]
@shefty Do you have any thoughts why the DCO checker fails? All commits was sighed off by commits' authors and me. OFI "Signed-off-by checker" is OK with the PR.

**Comment by shefty** [comment]
> @shefty <https://github.com/shefty>  Do you have any thoughts why the
> DCO checker fails? All commits was sighed off by commits' authors and
> me. OFI "Signed-off-by checker" is OK with the PR.

I think the DCO checker validates the author with the signed-off-by line and looks for a mismatch.  We can ignore this error if everything is signed off on.

**Comment by gladkovdmitry17** [comment]
> I think it's complaining about 4f8885c -- it was committed by Jeff Hammond <jeff.science@gmail.com>, but signed by Jeff Hammond <jeff.r.hammond@intel.com>. As humans, we know that's the same person. But the DCO checker rightfully flagged that as an incorrect signoff.

@jsquyres Thank you!
I've updated the PR and excluded the changes related to Travis testing of MLX provider. Seems our Travis is in not good shape with these changes ("before_install" section of script that should build the UCX library, seems is executed twice. The second attempt fails, because can't find `libm` library). I'm going to folow up this issue in separate PR (also we should wait when UCX crew fix build issue with clang compiler)

**Pull request merged**